app/models/url.rb

```diff
@@ -0,0 +1,21 @@
+require "uri"
+class Url < ActiveRecord::Base
+  validates :long, :uniqueness => true, format: { with: /\A#{URI::regexp}\z/}
+  before_save :convert
+
+  private
+
+  def convert
+     range = [*'0'..'9', *'a'..'z', *'A'..'Z']
```

Comment:
I'd put the `range` initializer inside the `unless` block, so you can avoid creating the array if you don't need to.